### PR TITLE
Inject Locale on Localization tests

### PR DESCRIPTION
**Pull request by santicarmo31**
## Motivation:
*  Some unit tests were failing due to its dependency on the system Locale

## Modifications:
* Inject Locale dependency to avoid unit test flakiness

## Result:
Screenshot 2025-01-07 at 6 33 54 PM

### CODE REVIEW

File: Tests/CurrencyTests/CurrencyValue+StringRepresentationTests.swift [hunk 1]
```diff
@@ -174,6 +175,7 @@ extension CurrencyValueStringRepresentationTests {
 
     let pounds = GBP(14928.018)
     formatter.currencyCode = GBP.alphabeticCode
+    formatter.locale = .init(identifier: "en_US")
```
Comment (Mordil) [inline Tests/CurrencyTests/CurrencyValue+StringRepresentationTests.swift:183]: ❕ Let's move this line up with the other formatter details at line 173/174

File: Tests/CurrencyTests/CurrencyValue+StringRepresentationTests.swift [hunk 2]
```diff
@@ -149,15 +150,15 @@ extension CurrencyValueStringRepresentationTests {
...
+    XCTAssertEqual("\(localize: USD(4321.389), for: Locale(identifier: "en_US"))", "$4,321.39")
   }
 }
 
 // MARK: LocalizedString
 
 extension CurrencyValueStringRepresentationTests {
   func test_localizedString_forLocale_usesDefaultLocale() {
-    XCTAssertEqual(USD(4321.389).localizedString(), "$4,321.39")
+    XCTAssertEqual(USD(4321.389).localizedString(for: .init(identifier: "en_US")), "$4,321.39")
```
Comment (Mordil) [inline Tests/CurrencyTests/CurrencyValue+StringRepresentationTests.swift]: 🤔 I'm honestly thinking we just delete this test + `test_stringInterpolation_forLocale_usesDefaultLocale` because we're effectively testing that the Swift language properly uses default values.

There IS value of having a test that asserts WHAT the default value is, but I think we need to rewrite the tests to basically duplicate the formatting logic with the `Locale.current` as the hard coded value, and then comparing the output

eg

```swift
let formatter = ...
formatter.locale = .current
let expectedOutput = formatter.string(for: ...)
XCTAssertEqual(...localizedString(), expectedOutput)
```

File: Tests/CurrencyTests/CurrencyValue+StringRepresentationTests.swift [hunk 3]
```diff
@@ -137,6 +137,7 @@ extension CurrencyValueStringRepresentationTests {
     formatter.numberStyle = .currency
     formatter.currencyGroupingSeparator = " "
     formatter.currencyDecimalSeparator = "'"
+    formatter.locale = .init(identifier: "en_US")
```
Comment (Mordil) [inline Tests/CurrencyTests/CurrencyValue+StringRepresentationTests.swift:140]: 🤦 Good work on this

File: Tests/CurrencyTests/CurrencyValue+StringRepresentationTests.swift (hunk 2 above)
Comment (Mordil) [reply Tests/CurrencyTests/CurrencyValue+StringRepresentationTests.swift]: Because as-is, this change of passing in a locale explicitly just duplicates other unit tests we already have

**Comment by Mordil** [comment]
@santicarmo31 Could you also update your PR description to match the format

```
Motivation:

Modifications:

Result:
```

File: Tests/CurrencyTests/CurrencyValue+StringRepresentationTests.swift [hunk 4]
```diff
@@ -149,15 +150,20 @@ extension CurrencyValueStringRepresentationTests {
...
 // MARK: LocalizedString
 
 extension CurrencyValueStringRepresentationTests {
   func test_localizedString_forLocale_usesDefaultLocale() {
-    XCTAssertEqual(USD(4321.389).localizedString(), "$4,321.39")
+    let formatter = NumberFormatter()
+    formatter.numberStyle = .currency
+    formatter.locale = .current
+    formatter.currencyCode = USD.alphabeticCode
+    let expectedOutput = formatter.string(for: Decimal(4321.389))
```
Comment (Mordil) [inline Tests/CurrencyTests/CurrencyValue+StringRepresentationTests.swift:167]: All this needs to be duplicated for `test_stringInterpolation_forLocale_usesDefaultLocale`

**Review by Mordil — APPROVED** [review]
